connectors/jdbc/translator-jdbc/src/main/java/org/teiid/translator/jdbc/JDBCMetadataProcessor.java

```diff
@@ -180,6 +181,16 @@ public void getConnectorMetadata(Connection conn, MetadataFactory metadataFactor
 			}
 		}
 		
+		if (this.schemaName != null) {
+			String escapeStr = metadata.getSearchStringEscape();
+			String escapedSchemaName = this.schemaName;
+			if (escapeStr != null) {
+				escapedSchemaName = escapedSchemaName.replace("_", escapeStr+"_");
+				escapedSchemaName = escapedSchemaName.replace("%", escapeStr+"%");
```

Comment:
The escapeStr will need escaped here as well.  Otherwise yes this looks good.

Reply:
ah, ofcourse. I will add that add push it. Thanks.